crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_assign.rs

```diff
@@ -0,0 +1,214 @@
+        let left_var_name = match left?.as_any_js_assignment()? {
+            AnyJsAssignment::JsComputedMemberAssignment(assignment) => assignment.text(),
+            AnyJsAssignment::JsIdentifierAssignment(assignment) => assignment.text(),
+            AnyJsAssignment::JsStaticMemberAssignment(assignment) => assignment.text(),
+            _ => return None,
+        };
+
+        let binary_expression = match node.right().ok()? {
+            AnyJsExpression::JsBinaryExpression(binary_expression) => binary_expression,
+            AnyJsExpression::JsParenthesizedExpression(param) => {
+                JsBinaryExpression::cast_ref(param.expression().ok()?.syntax())?
+            }
+            _ => return None,
+        };
+
+        let operator = binary_expression.operator().ok()?;
+        let shorthand_operator = get_shorthand(operator)?;
+        let is_commutative = is_commutative(operator);
+        let variable_position_in_expression =
+            parse_variable_reference_in_expression(&left_var_name, &binary_expression)?;
+
+        let replacement_expression = match variable_position_in_expression {
+            VariablePosition::Left => binary_expression.left().ok()?,
+            VariablePosition::Right => binary_expression.right().ok()?,
+        };
+
+        if !is_commutative && matches!(variable_position_in_expression, VariablePosition::Right) {
+            return None;
+        }
+
+        Some(RuleState {
+            shorthand_operator,
+            replacement_expression,
+        })
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        let shorthand_operator = state.shorthand_operator.to_string()?;
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "Assignment "<Emphasis>"(=)"</Emphasis>" can be replaced with operator assignment "<Emphasis>""{shorthand_operator}""</Emphasis>""
+            },
+        ))
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+
+        let mut mutation = ctx.root().begin();
+        let shorthand_operator = state.shorthand_operator;
+
+        let token_leading_trivia = node.operator_token().ok()?.leading_trivia().pieces();
+        let token_trailing_trivia = node.operator_token().ok()?.trailing_trivia().pieces();
+        let token = make::token(shorthand_operator)
+            .with_leading_trivia_pieces(token_leading_trivia)
+            .with_trailing_trivia_pieces(token_trailing_trivia);
+
+        let shorthand_node = node
+            .clone()
+            .with_operator_token_token(token)
+            .with_right(state.replacement_expression.clone());
+
+        mutation.replace_node(
+            AnyJsExpression::JsAssignmentExpression(node.clone()),
+            AnyJsExpression::JsAssignmentExpression(shorthand_node),
```

Comment:
`node` and `shorthand_node` have the same type? If yes, we should be able to skip the wrapping:
```suggestion
            node.clone(),
            shorthand_node,
```

Reply:
I was getting some weird error, so i used wrapping 😄

Your suggestion works fine